### PR TITLE
Configurable KV Table

### PR DESCRIPTION
**Pull request by sarahkemi**
This PR allows to the table name of the KV store to be configurable by the application using the gem, like so:

```
    GitHub::DS.configure do |config|
      config.table_name = "new_key_values"
    end
```

The gem will still maintain the default behavior of having the table name be `key_values` so current usage of the gem won't be broken, but it now allows for users of the gem to have flexibility in the name of the table in their database.

Would love thoughts and feedback on how I implemented the configurable table name. After the implementation is approved I still need to:
- [x] update README.md

### CODE REVIEW

File: lib/github/kv.rb [hunk 1]
```diff
@@ -71,6 +71,7 @@ def initialize(encapsulated_errors = [SystemCallError], use_local_time: false, &
       @encapsulated_errors = encapsulated_errors
       @use_local_time = use_local_time
       @conn_block = conn_block
+      @table_name = GitHub::SQL::LITERAL(GitHub::DS.config.table_name).value
```
Comment (jnunemaker) [inline lib/github/kv.rb]: I generally like this pattern for config and I've recently used it for the flipper gem. As I look at it though, the configuration for this class now feels like it is in two places -- initialize arguments and a config global. I'm curious if you have any thoughts on one of the following options:

* moving the other initialize options to config as well and changing initialize to receiving only config (with default to GitHub::DS.config)
* simply adding `table_name` as a parameter for `initialize` and dropping the new config

Comment (jnunemaker) [inline lib/github/kv.rb]: Is there any benefit or safety trade off to saving the `table_name` as a `GitHub::SQL::LITERAL` verse interpolating it like other bind variables?

File: test/github/ds_test.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require "test_helper"
+
+class GitHub::DSTest < Minitest::Test
+  def test_configures_correctly
+    GitHub::DS.configure do |config|
+      config.table_name = "example_key_values"
+    end
+
+    assert_equal GitHub::DS.config.table_name, "example_key_values"
+    GitHub::DS.reset
```
Comment (jnunemaker) [inline test/github/ds_test.rb]: Is there any place we could put this reset in addition to here to ensure it runs before every test?

Comment (sarahkemi) [reply test/github/ds_test.rb]: Placed it in appropriate `setup` methods!

File: lib/github/kv.rb (hunk 1 above)
Comment (sarahkemi) [reply lib/github/kv.rb]: Yeah I don't think it's necessary so I've removed it

Comment (sarahkemi) [reply lib/github/kv.rb]: I want to keep `table_name` in a config as it's used in other places outside of `GitHub::KV` so I went with option 1 😄

**Comment by sarahkemi** [comment]
👋 Would love more feedback or an approval on this! I understand that changes probably won't go out until after the holidays though 😅

**Comment by zerowidth** [comment]
Released as `v0.4.0`.

I suppose we could commit to a 1.0 version at some point...

**Pull request merged**